compiler/src/main/kotlin/dev/zacsweers/metro/compiler/Symbols.kt

```diff
@@ -501,6 +504,10 @@ internal class Symbols(
     val doubleCheckCompanionObject by lazy { doubleCheck.owner.companionObject()!!.symbol }
     val doubleCheckProvider by lazy { doubleCheckCompanionObject.requireSimpleFunction("provider") }
 
+    // SingleCheck support for unscoped bindings (lighter weight than DoubleCheck)
+    // Returns null if SingleCheck is not available in the runtime
+    abstract fun singleCheckProviderOrNull(): IrSimpleFunctionSymbol?
```

Comment:
Metro explicitly doesn't support single-check, was this necessary? I'd like to minimize scope creep since this already going to be a large set of changes